lib/src/model.dart

```diff
@@ -477,6 +482,24 @@ abstract class ReportSectionId {
   static const dependency = 'dependency';
   static const documentation = 'documentation';
   static const platform = 'platform';
+  static const nullSafety = 'null-safety';
+}
+
+enum ReportStatus {
+  @JsonValue('failed')
+  failed,
+  @JsonValue('partial')
+  partial,
+  @JsonValue('passed')
+  passed,
+}
+
+ReportStatus summarizeStatuses(Iterable<ReportStatus> ss) {
+  return ss.fold(ReportStatus.passed, minStatus);
+}
+
+ReportStatus minStatus(ReportStatus a, ReportStatus b) {
```

Comment:
nit: this code is not obvious, neat, but... could it have a comment?

Reply:
Done